### ISSUE TITLE
cycling syntax: support for cycling between specified dates/offsets from ICP

### ISSUE THREAD

**Issue by arjclark** [cylc/cylc#1601]
It would be useful if we could write recurrences along the lines of "repeat 6 hourly from the initial cycle point up to 30hrs after that".

To achieve that we can currently write:

```
  [[[R5/PT6H]]]
```

But it feels like I should be able to do something like:

```
[[[R/PT6H/+PT30H]]]
```

Similarly, it would be useful to be able to do specific repetitions between two specific dates:

```
[[[<date1>/<repetitions>/<date2>]]]
```

**Comment by arjclark** [comment]
Comments from @benfitzpatrick via email:

No, there isn't - but it would be nice:
```
[[[R/PT6H/^+PT30H]]]
```
would be OK if it worked (^ meaning initial cycle point, otherwise final cycle point would be used).

This:
```
[[[<date1>/<repetitions>/<date2>]]]
```
isn't really supported (there is an ISO 8601 recurrence form that kind of does this, but it isn't implemented right)

**Comment by benfitzpatrick** [comment]
Second comment references metomi/isodatetime#45

**Comment by oliver-sanders** [comment]
`[[[R/PT6H/^+PT30H]]]` has worked as of #1862.

**Comment by oliver-sanders** [comment]
`[[[<date1>/<repetitions>/<date2>]]]` Isn't supported but `[[[<repetitions>/<date1>/<date2>]]]` is.

Example (`initial cycle point = 2000`):
- `[[[R5/^/^+P4D]]]`
  - 20000101T00
  - 20000102T00
  - 20000103T00
  - 20000104T00

Will close this if @arjclark is happy.

**Comment by arjclark** [comment]
@oliver-sanders - can you ensure the examples you've got are documented in the User Guide so it's easy to find in future. Once that's done you can close.

**Comment by oliver-sanders** [comment]
This is now a documentation issue so I have merged it with #1531.
